### PR TITLE
EZP-30687: As a Maintainer I want to deprecated code in EzSystems\EzPlatformAdminUi\UserSetting be dropped

### PR DESCRIPTION
**Pull request by adamwojs**
| Question      | Answer
| ------------- | ---
| Tickets       |  [EZP-30687](https://jira.ez.no/browse/EZP-30687) 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

The second part of https://github.com/ezsystems/ezplatform-admin-ui/pull/1022.

#### Checklist:
- [ ] ~Implement tests~
- [X] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

**Comment by ViniTou** [comment]
I think we can also fix circular dependencies now between this and adminUI

File: src/lib/UserSetting/FormMapperInterface.php [hunk 1]
```diff
@@ -8,8 +8,20 @@
...
-use EzSystems\EzPlatformAdminUi\UserSetting as AdminUiUserSettings;
+use Symfony\Component\Form\FormBuilderInterface;
 
-interface FormMapperInterface extends AdminUiUserSettings\FormMapperInterface
+interface FormMapperInterface
 {
+    /**
+     * Creates 'value' form type representing editing control for setting user preference value.
+     *
+     * @param \Symfony\Component\Form\FormBuilderInterface $formBuilder
```
Comment (mikadamczyk) [inline src/lib/UserSetting/FormMapperInterface.php]: Do we need those `@params` and `@return`?

File: src/lib/UserSetting/ValueDefinitionInterface.php [hunk 2]
```diff
@@ -8,14 +8,41 @@
...
  * User Preferences are not displayed by default unless
  * ValueDefinitionInterface implementation is provided.
  */
-interface ValueDefinitionInterface extends AdminUiUserSettings\ValueDefinitionInterface
+interface ValueDefinitionInterface
 {
+    /**
+     * Returns name of a User Preference displayed in UI.
+     *
+     * @return string
```
Comment (mikadamczyk) [inline src/lib/UserSetting/ValueDefinitionInterface.php]:
```suggestion
```

File: src/lib/UserSetting/ValueDefinitionInterface.php [hunk 3]
```diff
@@ -8,14 +8,41 @@
...
+     */
+    public function getName(): string;
+
+    /**
+     * Returns description of a User Preference displayed in UI.
+     *
+     * @return string
```
Comment (mikadamczyk) [inline src/lib/UserSetting/ValueDefinitionInterface.php]:
```suggestion
```

File: src/lib/UserSetting/ValueDefinitionInterface.php [hunk 4]
```diff
@@ -8,14 +8,41 @@
...
+     */
+    public function getDescription(): string;
+
+    /**
+     * Returns formatted value to be displayed in UI.
+     *
+     * @param string $storageValue
```
Comment (mikadamczyk) [inline src/lib/UserSetting/ValueDefinitionInterface.php]:
```suggestion
```

File: src/lib/UserSetting/ValueDefinitionInterface.php [hunk 5]
```diff
@@ -8,14 +8,41 @@
...
+     *
+     * @return string
```
Comment (mikadamczyk) [inline src/lib/UserSetting/ValueDefinitionInterface.php]:
```suggestion
```

File: src/lib/UserSetting/ValueDefinitionInterface.php [hunk 6]
```diff
@@ -8,14 +8,41 @@
...
+     */
+    public function getDisplayValue(string $storageValue): string;
+
+    /**
+     * Returns default value for User Preference if none is defined.
+     *
+     * @return string
```
Comment (mikadamczyk) [inline src/lib/UserSetting/ValueDefinitionInterface.php]:
```suggestion
```

**Pull request merged**